indexFormat/implementation.md

```diff
@@ -0,0 +1,87 @@
+
+## LSIF exporters
+
+| Language | Repository |
+|--|--|
+| TypeScript/JavaScript | [lsif-node](https://github.com/Microsoft/lsif-node) |
+| Java | |
+| C# | |
+
+> Are we missing an implementation? File a new issue on GitHub to add it here.
+
+## LSIF exporter skeleton
+
+As [detailed in the spec](specification.md#project-exports-and-external-imports), the LSIF exporter consists of two tools: the index exporter and the package linker.
+
+### Index exporter
+
+The index exporter generates an LSIF dump for a workspace by traversing through source files and storing LSP responses. For TypeScript/JavaScript, [`lsif-tsc`](https://github.com/Microsoft/lsif-node/tree/master/tsc) is the index exporter.
+
+### Package linker
+
+The package linker converts the LSIF output of the index exporter into a global friendly index. By using package metadata, export `moniker` vertices are linked to packages available on a registry. For instance, the `observable` export from the mobx dependency is linked to the mobx dependency available on NPM. The package metadata is used to create the `packageInformation` vertices that reference external packages.
+
+For TypeScript/JavaScript, [`lsif-npm`](https://github.com/Microsoft/lsif-node/tree/master/npm) is the package manager linker for NPM.
+
+## Testing and validation
+
+### LSIF validation utility
+
+The [`lsif-util`](https://github.com/jumattos/lsif-util) tool can validate your generated LSIF output. Additionally, the tool can also be used to search the output and visualize via Graphviz.
+
+### VS Code LSIF extension
+
+With the [LSIF extension for VS Code](https://github.com/Microsoft/vscode-lsif-extension), you can dogfood an LSIF index to power navigation inside VS Code.
+
+## Recommended patterns
+
+We have seen the following patterns work well in existing implementations.
+
+### Method checklist
+
+For an ideal integration with Rich Code Navigation, the following methods are required. For some languages, methods such as `textDocument/declaration` might not be applicable.
+
+- [ ] `textDocument/hover`
+- [ ] `textDocument/definition`
+- [ ] `textDocument/references`
+- [ ] `textDocument/implementation`
+- [ ] `textDocument/declaration`
+- [ ] `textDocument/typeDefinition`
+- [ ] `textDocument/diagnostic`
+- [ ] Cross-repo navigation for dependencies
+
+### Cross-platform
+
+If the LSIF exporter does not work across platforms (Windows, Linux, Mac), platform dependencies should be called out.
+
+### `--outputFormat` execution flag
+
+It is recommended that the LSIF exporter support two output formats: `line` and `json`. The `line` format is required for an integration with Rich Code Navigation.
+
+1. `line`: Output is a series of JSON objects (vertex or edge) separated by newline. This format is optimized for streaming output, and works better for larger repos.
+1. `json`: Output is a valid JSON array, with each JSON object (vertex or edge) as an element of this array. The VS Code LSIF extension uses this format.
```

Comment:
I would remove `json`. the only format we should support is line. A json outputer can be a pipe tool as well.